### PR TITLE
Rename config for enforcing a minimum number of label matchers

### PR DESCRIPTION
**Pull request by salvacorts**
**What this PR does / why we need it**:
Followup PR for https://github.com/grafana/loki/pull/8918 renaming config. See https://github.com/grafana/loki/pull/8918/files#r1151820792.

**Which issue(s) this PR fixes**:
Fixes https://github.com/grafana/loki-private/issues/699

**Special notes for your reviewer**:

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Review by jeschkies — APPROVED** [review]

File: pkg/validation/limits.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ type Limits struct {
 	BlockedQueries []*validation.BlockedQuery `yaml:"blocked_queries,omitempty" json:"blocked_queries,omitempty"`
 
 	RequiredLabels       []string `yaml:"required_label_matchers,omitempty" json:"required_label_matchers,omitempty" doc:"description=Define a list of required selector labels."`
-	RequiredNumberLabels int      `yaml:"required_number_label_matchers,omitempty" json:"required_number_label_matchers,omitempty" doc:"description=Minimum number of label matchers a query should contain."`
+	RequiredNumberLabels int      `yaml:"minimum_labels_number,omitempty" json:"required_number_label_matchers,omitempty" doc:"description=Minimum number of label matchers a query should contain."`
```
Comment (DylanGuedes) [inline pkg/validation/limits.go:173]:
```suggestion
	RequiredNumberLabels int      `yaml:"minimum_labels_number,omitempty" json:"minimum_labels_number,omitempty" doc:"description=Minimum number of label matchers a query should contain."`
```

**Review by DylanGuedes — DISMISSED** [review]

File: pkg/validation/limits.go (hunk 1 above)
Comment (salvacorts) [reply pkg/validation/limits.go:173]: Thank buddy! I need to stop working on two things at the same time 😅

Comment (jeschkies) [reply pkg/validation/limits.go]: We didn't rename it for `required_label_matchers` either :see_no_evil:

**Review by jeschkies — APPROVED** [review]

File: pkg/validation/limits.go (hunk 1 above)
Comment (salvacorts) [reply pkg/validation/limits.go]: Done :)

**Pull request merged**